lib/ecto/adapters/clickhouse/structure.ex

```diff
@@ -61,10 +61,13 @@ defmodule Ecto.Adapters.ClickHouse.Structure do
     table = @conn.quote_table(database, table)
     stmt = "SELECT * FROM #{table} FORMAT Values"
 
-    with {:ok, %{rows: rows}, conn} <- exec(conn, stmt) do
+    with {:ok, %{rows: [rows]}, conn} <- exec(conn, stmt) do
```

Comment:
Technically we can get more than one row in results for tables with lots of rows in schema_migrations table. I think ClickHouse would start splitting it at ~50k characters, it's a 

We'd also need to handle the error case explicitly since we are adding an `else`.

```elixir
else
  {:ok, %{rows: []}, conn} -> []
  {:error, _reason} = error -> error # or was it {:error, _reason, conn}, I don't remember
end
```

Reply:
I don't think it is a problem if there are multiple rows:
```
iex(1)> with [x] <- [1,2,3,4] do
...(1)>   x
...(1)> end
[1, 2, 3, 4]
iex(2)>
```

Reply:
```elixir
iex(1)> with [x] <- [1,2,3], do: true
[1, 2, 3]
```

Reply:
In `with`, there is an implicit `else` block that returns the unmatched values.